src/hotspot/cpu/riscv/c1_Runtime1_riscv.cpp

```diff
@@ -925,6 +925,51 @@ OopMapSet* Runtime1::generate_code_for(C1StubId id, StubAssembler* sasm) {
       }
       break;
 
+    case C1StubId::is_instance_of_id:
+      {
+        // Mirror: x10
+        // Object: x11
+        // Temps: x13, x14, x15, x16, x17
+        // Result: x10
+
+        // Get the Klass* into x16
+        Register klass = x16, obj = x11, result = x10;
+        __ ld(klass, Address(x10, java_lang_Class::klass_offset()));
+
+        Label fail, is_secondary, success;
+
+        __ beqz(klass, fail); // Klass is null
+        __ beqz(obj, fail); // obj is null
+
+        __ lwu(x13, Address(klass, in_bytes(Klass::super_check_offset_offset())));
+        __ mv(x17, in_bytes(Klass::secondary_super_cache_offset()));
+        __ beq(x13, x17, is_secondary); // Klass is a secondary superclass
+
+        // Klass is a concrete class
+        __ load_klass(x15, obj);
+        __ add(x17, x15, x13);
+        __ ld(x17, Address(x17));
+        __ beq(klass, x17, success);
+        __ j(fail);
```

Comment:
Hmm, I think this jump could be saved if we put a direct return here instead. Like:
```
        __ beq(klass, x17, success);
        __ mv(result, 0);
        __ ret();
```
What do you think? @Hamlin-Li @zifeihan

Reply:
Yes, I think we can put a direct return here.

Reply:
Yes, seems better.
I was thinking to move bind(fail) here, but seems current lookup_secondary_supers_table_var only accept a success label.